### DISCUSSION TITLE
can kiwix officially support its own flatpak

### DISCUSSION THREAD

**Discussion by RustoMCSpit** [kiwix/kiwix-desktop#1160] [Ideas]
currently it is unofficially packaged by someone else

**Comment by kelson42** [comment]
It's not true, we do our own flatpak and distribute it. Look at  https://download.kiwix.org/release/kiwix-desktop/.

**Comment by RustoMCSpit** [comment]
![image](https://github.com/user-attachments/assets/485c9cb6-373f-4dc7-9de2-6dc46ac18e20)
![image](https://github.com/user-attachments/assets/4273b91d-680d-4076-9d07-bc2d151e5767)


then you need to verify it on flathub as it shows up as a non-official package

**Reply by RustoMCSpit** [reply]
![image](https://github.com/user-attachments/assets/fc6ef6b2-2959-47ea-b51f-3a42b1ae0fa0)
youre also not "safe"

**Comment by Efenstor** [comment]
For those who are worried, this issue was mostly fixed.

The Flathub release is now verified:

<img width="1044" height="680" alt="Screenshot at 2025-10-09 13-56-08" src="https://github.com/user-attachments/assets/7ec08447-8c73-4864-9aba-614c4c2f95cc" />

As of the critical unsafe features, only the outdated runtimes remain. I hope it will be fixed when the new release finally comes.

<img width="1222" height="1248" alt="Screenshot at 2025-10-09 13-56-20" src="https://github.com/user-attachments/assets/18e75bfd-ec82-4fd2-a381-0420b1e6d4cc" />

Network access is needed for downloading ZIM files from the Kiwix library, and as far as I know audio is needed for playing audio clips in the articles (and probably also for TTS).

**Reply by kelson42** [reply]
@Efenstor We will make a new release soon. I'm glad if you can vet our nihtly build at https://download.kiwix.org/nightly/2025-09-29/org.kiwix.desktop.2025-09-29.flatpak

**Reply by Efenstor** [reply]
Right off the bat I cannot run it :) but the issue is specific only for my particular configuration: I am using the custom [Uralic Universal](https://github.com/Efenstor/Uralic-keyboard-layouts) keyboard layout for my work in linguistics.

> Gtk-Message: 15:06:52.094: Failed to load module "xapp-gtk3-module"
> Qt: Session management error: Could not open network socket
> QApplication: invalid style override 'kvantum' passed, ignoring it.
> 	Available styles: Breeze, Windows, Fusion
> Path override failed for key base::DIR_APP_DICTIONARIES and path '/app/bin/qtwebengine_dictionaries'
> xkbcommon: ERROR: Couldn't process include statement for 'ru(uralic_universal)'
> xkbcommon: ERROR: Abandoning symbols file "(unnamed)"
> xkbcommon: ERROR: Failed to compile xkb_symbols
> xkbcommon: ERROR: Failed to compile keymap
> [2:37:1009/150652.581313:FATAL:xkb_keyboard_layout_engine.cc(654)] Keymap file failed to load: us,ru-,uralic_universal

What's interesting is that I never had this issue with any other Flatpak app and the old Kiwix release as well. If I replace the Uralic Universal layout with any standard layout it starts without issue. Adding /usr/share/X11/xkb to the filesystem overrides is not allowed as /usr is "reserved" by Flatpak. But I totally need that layout for my work so I'll try to do something about it.

Another unresolved issue is that the startup time is still about half a minute on my not-so-bad system (32 GB RAM, 8 core AMD Ryzen 7). I wonder why the Android version starts so quickly and generally much more responsive than the desktop version?

I'll do more tests of the rest of the functionality, of course, as soon as I'll have some time.

And another thing: please review the [pull requests](https://github.com/kiwix/kiwix-desktop/pulls), there is a lot of useful features and fixes contributed by the people, many of these are waiting there for almost a year.
